src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -123,10 +124,11 @@ private ParseFormulasResult ParseFormulas(string script)
                             // Parse expression
                             var result = ParseExpr(Precedence.None);
 
-                            namedFormulas.Add(new KeyValuePair<IdentToken, TexlNode>(thisIdentifier.As<IdentToken>(), result));
+                            namedFormulas.Add(new NamedFormula(thisIdentifier.As<IdentToken>(), result, offset));
                         }
 
                         _curs.TokMove();
+                        offset = _curs.CurrentCharIndex;
```

Comment:
consider setting the offset at the call to ParseExpr, that way the token starting at 0 is the initial token of the actual rule.

Reply:
that way it'll match non-namedformula tokens within a single expression